backends/gdx-backend-lwjgl3/src/com/badlogic/gdx/backends/lwjgl3/Lwjgl3AWTCanvas.java

```diff
@@ -0,0 +1,541 @@
+//					Gdx.gl.glClear(GL20.GL_COLOR_BUFFER_BIT);
+					Gdx.gl.glViewport(0,0,getWidth(), getHeight());
+//					Gdx.gl.glEnable(GL20.GL_SCISSOR_TEST);
+//					Gdx.gl.glScissor(0,0,getWidth(),getHeight());
+					try {
+						boolean systemPaint = !(EventQueue.getCurrentEvent() instanceof NonSystemPaint);
+						Lwjgl3AWTCanvas.this.render(systemPaint);
+						Toolkit.getDefaultToolkit().getSystemEventQueue().postEvent(nonSystemPaint);
+					} catch (Throwable ex) {
+						exception(ex);
+					}
+					swapBuffers();
+				}
+			};
+		} catch (Throwable ex) {
+			exception(ex);
+			return;
+		}
+
+		canvas.setBackground(new Color(config.initialBackgroundColor.r, config.initialBackgroundColor.g,
+			config.initialBackgroundColor.b, config.initialBackgroundColor.a));
+
+		graphics = new Lwjgl3AWTGraphics(this) {
+			@Override
+			public void setTitle (String title) {
+				super.setTitle(title);
+				Lwjgl3AWTCanvas.this.setTitle(title);
+			}
+
+			@Override
+			public boolean setWindowedMode (int width, int height) {
+				if (!super.setWindowedMode(width, height)) return false;
+				Lwjgl3AWTCanvas.this.setDisplayMode(width, height);
+				return true;
+			}
+
+			@Override
+			public boolean setFullscreenMode (Graphics.DisplayMode displayMode) {
+				if (!super.setFullscreenMode(displayMode)) return false;
+				Lwjgl3AWTCanvas.this.setDisplayMode(displayMode.width, displayMode.height);
+				return true;
+			}
+
+//			public boolean shouldRender () {
+//				synchronized (this) {
+//					boolean rq = requestRendering;
+//					requestRendering = false;
+//					return rq || isContinuous;
+//				}
+//			}
+		};
+
+		if (!config.disableAudio && Gdx.audio == null) audio = new OpenALLwjgl3Audio();
+		if (Gdx.files == null) files = new Lwjgl3Files();
+		if (Gdx.net == null) net = new Lwjgl3Net(config);
+		input = new Lwjgl3AWTInput(this);
+		setGlobals();
+	}
+
+	protected void setDisplayMode (int width, int height) {
+	}
+
+	protected void setTitle (String title) {
+	}
+
+	@Override
+	public ApplicationListener getApplicationListener () {
+		return listener;
+	}
+
+	public AWTGLCanvas getCanvas () {
+		return canvas;
+	}
+
+	@Override
+	public Audio getAudio () {
+		return Gdx.audio;
+	}
+
+	@Override
+	public Files getFiles () {
+		return files;
+	}
+
+	@Override
+	public Graphics getGraphics () {
+		return graphics;
+	}
+
+	@Override
+	public Input getInput () {
+		return input;
+	}
+
+	@Override
+	public Net getNet () {
+		return net;
+	}
+
+	@Override
+	public ApplicationType getType () {
+		return ApplicationType.Desktop;
+	}
+
+	@Override
+	public int getVersion () {
+		return 0;
+	}
+
+	void setGlobals () {
+		Gdx.graphics = graphics;
+		Gdx.gl30 = graphics.getGL30();
+		Gdx.gl20 = Gdx.gl30 != null ? Gdx.gl30 : graphics.getGL20();
+		Gdx.gl = Gdx.gl30 != null ? Gdx.gl30 : Gdx.gl20;
+
+		Gdx.app = this;
+		if (audio != null) Gdx.audio = audio;
+		if (files != null) Gdx.files = files;
+		if (net != null) Gdx.net = net;
+		Gdx.graphics = graphics;
```

Comment:
Is redundant.

Reply:
yeah, I copy pasted it from both LwjglCanvas from lwjgl2 and Lwjgl3Window 😆